crates/router/src/core/webhooks/outgoing.rs

```diff
@@ -546,23 +551,49 @@ fn get_webhook_url_from_business_profile(
         .map(ExposeInterface::expose)
 }
 
-pub(crate) fn get_outgoing_webhook_request(
+pub(crate) async fn get_outgoing_webhook_request(
     merchant_account: &domain::MerchantAccount,
     outgoing_webhook: api::OutgoingWebhook,
-    payment_response_hash_key: Option<&str>,
+    business_profile: &diesel_models::business_profile::BusinessProfile,
+    key_store: &domain::MerchantKeyStore,
 ) -> CustomResult<OutgoingWebhookRequestContent, errors::WebhooksFlowError> {
     #[inline]
-    fn get_outgoing_webhook_request_inner<WebhookType: types::OutgoingWebhookType>(
+    async fn get_outgoing_webhook_request_inner<WebhookType: types::OutgoingWebhookType>(
         outgoing_webhook: api::OutgoingWebhook,
-        payment_response_hash_key: Option<&str>,
+        business_profile: &diesel_models::business_profile::BusinessProfile,
+        key_store: &domain::MerchantKeyStore,
     ) -> CustomResult<OutgoingWebhookRequestContent, errors::WebhooksFlowError> {
         let mut headers = vec![(
             reqwest::header::CONTENT_TYPE.to_string(),
             mime::APPLICATION_JSON.essence_str().into(),
         )];
 
         let transformed_outgoing_webhook = WebhookType::from(outgoing_webhook);
-
+        let payment_response_hash_key = business_profile.payment_response_hash_key.clone();
+        let custom_headers = decrypt::<serde_json::Value, masking::WithType>(
+            business_profile
+                .outgoing_webhook_custom_http_headers
+                .clone(),
+            key_store.key.get_inner().peek(),
+        )
```

Comment:
We'd preferably want to have this implementation as part of domain type implementation rather than encrypting/decrypting it individually in separate locations.

Reply:
Sure will take this in separate pr https://github.com/juspay/hyperswitch-cloud/issues/5984